tests/Mafin.Web.Api.Rest.Tests.Unit/MafinHttpClientBuilderTests.cs

```diff
@@ -36,33 +38,33 @@ public void WithAuthHandler_WhenHandlerPassed_ShouldDelegateHttpCallToSameHandle
     {
         const string sendMethodName = "Send";
 
-        Mock<HttpClientHandler> authHandlerMock = new();
-        Mock<HttpRequestMessage> requestMock = new();
+        var authHandlerMock = Substitute.For<HttpClientHandler>();
+        var requestMock = Substitute.For<HttpRequestMessage>();
+        var responseMock = Substitute.For<HttpResponseMessage>();
 
-        authHandlerMock.Protected()
-            .Setup<HttpResponseMessage>(sendMethodName, requestMock.Object, ItExpr.IsAny<CancellationToken>())
-            .Returns(new Mock<HttpResponseMessage>().Object)
-            .Verifiable();
+        typeof(HttpClientHandler).InvokeMember(sendMethodName, BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.InvokeMethod, null, authHandlerMock, new object[] { requestMock, Arg.Any<CancellationToken>() }).Returns(responseMock);
 
         _builder = new MafinHttpClientBuilder(Url);
-        var client = _builder.WithAuthHandler(authHandlerMock.Object).Build();
+        var client = _builder.WithAuthHandler(authHandlerMock).Build();
 
-        _ = client.Send(requestMock.Object);
+        _ = client.Send(requestMock);
 
-        authHandlerMock.Protected().Verify(sendMethodName, Times.Once(), requestMock.Object, ItExpr.IsAny<CancellationToken>());
+        Received.InOrder(() =>
+        {
+            typeof(HttpClientHandler).InvokeMember(sendMethodName,BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.InvokeMethod, null, authHandlerMock, new object[] { requestMock, Arg.Any<CancellationToken>() });
```

Comment:
Please, fix the warnings in this line.
Also, I see it being used twice.

Do you think it's possible to extract it, for example to a local function?

Reply:
Fixed